### PR TITLE
Enable invalid cert test

### PR DESCRIPTION
**Pull request by ngastelum-ms**

### CODE REVIEW

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml
Comment (timtay-microsoft) [inline vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:141]: These tests will run for android, windows, and linux builds, FYI. Do these steps enable the invalid cert server for those builds as well? Or do we need to duplicate these steps in each of the windows/linux/android builds?

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/ProvisioningTests.java
Comment (schoims) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/ProvisioningTests.java:92]: You can remove import org.junit.Ignore;

File: iot-e2e-tests/docker/readme.md
Comment (schoims) [inline iot-e2e-tests/docker/readme.md:9]:
```suggestion
- Install WSL (See installation details here: https://docs.microsoft.com/windows/wsl/install). Reboot after install.
```

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml
Comment (ngastelum-ms) [reply vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:141]: Good catch, right now it seems this only works on Linux due to some issue with Docker on Windows.

Comment (andyk-ms) [reply vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:141]: Linux is the only one this will work for now (and we should test locally how well it works for android - it should work but this will be the first we test it). Docker for Windows have a bug that is reported but have no resolution and therefore windows agent in the pipeline will not work. On our local test, there are instruction to use WSL (linux) to run this on our windows laptop.

Also, there is a build.ps script for .NET SDK that only run invalid cert tests (Traits) when it is linux (non-window agents). There isn't an equivalent for that here in the changes.

File: iot-e2e-tests/docker/readme.md
Comment (timtay-microsoft) [inline iot-e2e-tests/docker/readme.md:5]: Links to the C# SDK repo?

File: vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml
Comment (ngastelum-ms) [inline vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml:180]: Script seems to not work on android as is, so this is currently only for the linux builds

File: iot-e2e-tests/docker/readme.md
Comment (patilsnr) [inline iot-e2e-tests/docker/readme.md:7]: nit: is it worth specifying which version of WSL to use? I think the default these days is 2 but I'm not entirely sure.

File: iot-e2e-tests/docker/readme.md
Comment (patilsnr) [inline iot-e2e-tests/docker/readme.md:9]: nit: for consistency, would it be worth using hyperlinks?

File: iot-e2e-tests/docker/readme.md
Comment (patilsnr) [inline iot-e2e-tests/docker/readme.md:22]: `... not  in WSL).`

File: iot-e2e-tests/docker/readme.md
Comment (patilsnr) [inline iot-e2e-tests/docker/readme.md:32]: missing periods on this and next point

**Review by patilsnr — APPROVED** [review]
Looks good! I don't know enough about `.cfg` files to give a full judge of what's going on so I would wait for an approval from someone who does before merging.

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/MessagingClientTests.java
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/MessagingClientTests.java:339]: Consider moving this to Tools.java where we have a method called "isAndroid" that is similar to what you have here.

```
public static boolean isLinux
{
    return System.getProperty("os.name").toLowerCase().contains("linux");
}
```

So that the test reads a bit more easily

```java
Assume.assumeTrue(Tools.isLinux());
```

Comment (timtay-microsoft) [reply iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/MessagingClientTests.java:339]: It also cuts down on the duplicated code you currently have

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/ProvisioningTests.java
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/ProvisioningTests.java:67]: duplicated logic for how to check if the test is run on Linux. See my other comment about factoring this out somewhere

**Review by timtay-microsoft — APPROVED** [review]

File: iot-e2e-tests/docker/readme.md
Comment (timtay-microsoft) [reply iot-e2e-tests/docker/readme.md:9]: Good idea

**Review by schoims — APPROVED** [review]

File: iot-e2e-tests/docker/readme.md
Comment (ngastelum-ms) [reply iot-e2e-tests/docker/readme.md:7]: Looks like the default is wsl 2, I'm thinking it's not worth specifying since the learn docs we link to don't either except in the specific version section.

**Pull request merged**
